web/quiz_views.py

```
@@ -310,24 +601,50 @@
             )
-            return redirect("quiz_detail", quiz_id=quiz.id)
+            # Find the latest attempt to show results
+            latest_attempt = UserQuiz.objects.filter(quiz=quiz, user=user).order_by("-start_time").first()
+            return redirect("quiz_results", user_quiz_id=latest_attempt.id)
+
+    user_quiz = None
+    remaining_time = 0
+    # Check for existing incomplete attempts first
+    if user:
+        # calculate remaining time
+        def calculate_remaining_time():
```

Ruff (0.8.2)

Missing return type annotation for private function `calculate_remaining_time`

(ANN202)